esp-hal/src/i2s.rs

```diff
@@ -2378,13 +2335,31 @@ pub mod asynch {
         async fn read_dma_async(&mut self, words: &mut [u8]) -> Result<(), Error> {
             let (ptr, len) = (words.as_mut_ptr(), words.len());
 
-            self.rx_channel.listen_eof();
+            if len % 4 != 0 {
+                return Err(Error::IllegalArgument);
+            }
+
+            // Reset TX unit and TX FIFO
+            T::reset_rx();
 
-            self.start_rx_transfer_async(ptr, len, false)?;
+            let mut future = DmaRxFuture::new(&mut self.rx_channel);
 
-            DmaRxFuture::new(&mut self.rx_channel).await;
+            // configure DMA outlink
+            future
+                .rx()
+                .prepare_transfer_without_start(false, T::get_dma_peripheral(), ptr, len)?;
+            future.rx().start_transfer()?;
 
-            // ??? T::reset_tx();
+            // set I2S_TX_STOP_EN if needed
+
+            // start: set I2S_RX_START
+            #[cfg(not(esp32))]
+            T::rx_start(len - 1);
```

Comment:
Passing comment, could this be done inside the T impl for esp32, to avoid the cfg here or does that just move the cfg elsewhere?

Reply:
Naively it would just move to the common `RegisterAccess` for ESP32 and S2 but we can probably do something about that 👍

Reply:
Ended up with a single `cfg` in one place - it was too easy to forget about that (and I forgot it in one place before)